src/Toolkit/Toolkit.UI.Controls/SearchView/SearchView.cs

```diff
@@ -820,6 +744,135 @@ internal SuggestionsGrouped(ISearchSource owningSource, IEnumerable<SearchSugges
             System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => _suggestions.GetEnumerator();
         }
 #endif
+
+        /// <summary>
+        /// SearchViewTemplateSettings provides a set of properties that are used when you define a new control template for a control that derives from <see cref="SearchView"/>.
+        /// </summary>
+        public SearchViewTemplateSettings TemplateSettings
+        {
+#if WINDOWS_XAML
+            get => (SearchViewTemplateSettings)GetValue(TemplateSettingsProperty);
+#elif WPF
+            get => (SearchViewTemplateSettings)GetValue(TemplateSettingsPropertyKey.DependencyProperty);
+#endif
+        }
+
+#if WINDOWS_XAML
+        /// <summary>
+        /// Identifies the <see cref="TemplateSettings"/> dependency property.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static readonly DependencyProperty TemplateSettingsProperty =
+            DependencyProperty.Register(nameof(TemplateSettings), typeof(SearchViewTemplateSettings), typeof(SearchView), new PropertyMetadata(null));
+#elif WPF
+        internal static readonly DependencyPropertyKey TemplateSettingsPropertyKey =
+            DependencyProperty.RegisterReadOnly(nameof(TemplateSettings), typeof(SearchViewTemplateSettings), typeof(SearchView), new FrameworkPropertyMetadata());
+#endif
+    }
+
+    /// <summary>
+    /// SearchViewTemplateSettings provides a set of properties that are used when you define a new control template for a control that derives from <see cref="SearchView"/>.
+    /// </summary>
+    /// <remarks>
+    /// TemplateSettings properties are always intended to be used in XAML, not code. They are read-only sub-properties of a read-only TemplateSettings property of a parent control.
+    /// </remarks>
+    /// <seealso cref="SearchView.TemplateSettings"/>
+#if WINUI
+    [WinRT.GeneratedBindableCustomProperty]
+#endif
+    public partial class SearchViewTemplateSettings : INotifyPropertyChanged
+    {
+        private SearchView _owner;
+        internal SearchViewTemplateSettings(SearchView owner)
+        {
+            _owner = owner;
+        }
+
+        /// <summary>
+        /// Gets the visibility for the presentation of the <see cref="SearchView.NoResultMessage"/>.
+        /// </summary>
+        public Visibility ResultMessageVisibility
```

Comment:
This probably should've been named *No*ResultMessageVisibility, but I would understand if you don't want to mess with existing names as part of this PR